papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,563 @@
+Numerous other well-known Python packages (h5py and Zarr immediately come to mind) also do this in order to make working with complex resources feel as natural as possible to Python users.
+Most such packages implement this layer directly for their particular use case, but the nature of the problem suggested to us the possibility of developing a more generic representation of this interface.
+Indeed, the Python standard library's ``collections.abc`` module is designed exactly for the purpose of making it easy to define objects that "look like" standard Python objects while having completely customizable behavior under the hood.
+As such, we saw an opportunity to specialize this pattern for a specific use case: the transparent synchronization of a Python object with an underlying resource.
+
+The **synced collections** framework represents the culmination of our efforts in this direction, providing a generic framework within the interfaces of any abstract data type can be mapped to arbitrary underlying synchronization protocols.
+In **signac** this framework allows us to hide the details of a particular file storage medium (like JSON) behind a dictionary-like interface, but it can just as easily be used for tasks like creating a set-like interface to an underlying extension type or wrapping a directory manager in a list-like interface.
+This section will offer a high-level overview of this framework and our plans for its use within **signac**, with an eye to potential users in other domains as well.
+
+Summary of Features
+~~~~~~~~~~~~~~~~~~~
+
+We designed synced collections to be flexible, easily extensible, and independent of **signac**'s data model.
+Most practical use cases for this framework involve an underlying resource that may be modified by any number of associated in-memory objects that behave like standard Python collections such as dictionaries or lists.
+Therefore, all normal operations must be preceded by loading from this resource and updating the in-memory store, and they must be succeeded by a subsequent save to that resource.
+The central idea behind synced collections is to decouple this process into two distinct groups of tasks: the saving and loading of data from a particular resource backend, and the synchronization of two in-memory objects of a given type.
+This delineation allows us to, for instance, encapsulate all logic for JSON files into a single ``JSONCollection`` class and then combine it with dictionary- or list-like ``SyncedDict``/``SyncedList`` classes via inheritance to create fully functional JSON-backed dictionaries or lists.
+Such synchronization has significant performance implications, so the framework also exposes an API to implement buffering protocols to collect operations into a single transaction before submitting them to the underlying resource.
```

Comment:
```suggestion
Such synchronization has significant performance implications, so the framework also exposes an API to implement buffering protocols to collect operations into a single transaction before submitting them to the underlying resource.
```

"Implication" doesn't specify negative or positive

Reply:
@atravitz this does not have a suggestion in it.

Reply:
```suggestion
Such synchronization significantly lowers performance, so the framework also exposes an API to implement buffering protocols to collect operations into a single transaction before submitting them to the underlying resource.
```